src/microbe_stage/Microbe.Interior.cs

```diff
@@ -904,20 +904,17 @@ private void HandleReproduction(float delta)
 
         if (GameWorld.WorldSettings.LimitReproductionCompoundUseSpeed)
         {
-            remainingAllowedCompoundUse = Constants.MICROBE_REPRODUCTION_MAX_COMPOUND_USE * delta;
-
+            // TODO: make the current patch affect this?
+            remainingFreeCompounds = Constants.MICROBE_REPRODUCTION_FREE_COMPOUNDS * (HexCount * 0.02f + 1.0f) * delta;
             if (IsMulticellular)
-                remainingAllowedCompoundUse *= Constants.EARLY_MULTICELLULAR_REPRODUCTION_COMPOUND_MULTIPLIER;
+                remainingFreeCompounds *= Constants.EARLY_MULTICELLULAR_REPRODUCTION_COMPOUND_MULTIPLIER;
         }
 
         if (GameWorld.WorldSettings.PassiveGainOfReproductionCompounds)
         {
-            // TODO: make the current patch affect this?
-            // TODO: make surface area of the cell (and colony) affect this
-            remainingFreeCompounds = Constants.MICROBE_REPRODUCTION_FREE_COMPOUNDS * delta;
-
+            remainingAllowedCompoundUse = remainingFreeCompounds * Constants.MICROBE_REPRODUCTION_MAX_COMPOUND_USE;
             if (IsMulticellular)
-                remainingFreeCompounds *= Constants.EARLY_MULTICELLULAR_REPRODUCTION_COMPOUND_MULTIPLIER;
+                remainingAllowedCompoundUse *= Constants.EARLY_MULTICELLULAR_REPRODUCTION_COMPOUND_MULTIPLIER;
         }
```

Comment:
Moving these variables around now has incorrect logic regarding LimitReproductionCompoundUseSpeed and PassiveGainOfReproductionCompounds flags. This will have very unintended side effects if people play with only just one of these options enabled. I can fix the logic (and put the colony size TODO comment back), once the math has been finalized.

Reply:
I think I probably fixed this in my commit.